LocationSync/LocationSync.cpp

```diff
@@ -54,6 +73,12 @@ namespace Plugin {
         config.FromString(service->ConfigLine());
         string version = service->Version();
 
+        auto timeZone = readTextFile(TZ_FILE);
```

Comment:
Let please preserve backwards compatibility. If I read the code I think what you would like to do is overrule the TZ environment variable if it is stored in a file. 
So to make such a feature backwards compatible I suggest adding a config parameter here: https://github.com/rdkcentral/rdkservices/blob/sprint/2107/LocationSync/LocationSync.h#L157 and call it override. In the override you can specify a path (which helps you get ride of the magical, see comments above). 
Now suppose that the config parameter Override.IsSet() and it contains a real filename, only than you try to load that file and you set the environment variable here. This way existing customers will not face havoc if they would place a file at the hardcoded location "/opt/persistent/timeZoneDST"